### DISCUSSION TITLE
default in python class not set

### DISCUSSION THREAD

**Discussion by bert2002** [sqlalchemy/sqlalchemy#9867] [Usage Questions] [answered]
Hi,
I am using SQLAlchemy 2.0 with FastAPI and created my own tables. They are both looking like this:

```
class UserSettings(Base):
  __tablename__ = 'settings'
  id = Column(Integer, primary_key=True, autoincrement=True)
  user_id = Column(String(36))
  email = Column(String(64))
  created_at = Column(DateTime, default=datetime.now)
  updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now)

  def __init__(self, user_id, email):
    self.user_id = user_id
    self.email = email
```
and
```
class Pricing(Base):
  __tablename__ = 'pricing'
  id = Column(Integer, primary_key=True, autoincrement=True)
  coinid = Column(String(64))
  symbol = Column(String(64))
  name = Column(String(64))
  created_at = Column(DateTime, default=datetime.now)
  updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now)

  def __init__(self, coinid, symbol, name):
    self.coinid = coinid
    self.symbol = symbol
    self.name = name
```
Both tables getting created and I can fill it with data, but just in the `settings` table I can see that `created_at` and `updated_at` are created - not in `pricing`.

I added `nullable=False` for the `pricing` table, but then I am getting an error that the default is not set:
```
pymysql.err.OperationalError: (1364, "Field 'created_at' doesn't have a default value")
```

I cant see any difference on why this is not getting created. Any idea?

Thanks,
bert

**Comment by CaselIT** [comment]
Hi,

How are you getting that error? can you provide an example?

**Reply by bert2002** [reply, marked as answer]
Actually I found out someone is running raw sql queries and not using the python class to add or update records. Works as expected.

**Reply by CaselIT** [reply]
ok!
